### ISSUE TITLE
Relax: look at relax's math

### ISSUE THREAD

**Issue by vxlcoder** [CGCookie/retopoflow#514]
Relax will cause polygons to flip :(

![screen recording 2018-07-10 at 07 29 pm](https://user-images.githubusercontent.com/3790699/42545253-beeeccba-8484-11e8-8aaf-df76bcaeff80.gif)

**Comment by Sinhouse85** [comment]
@vxlcoder 
I have found the relax to occasionally jitter points around, destroying flow rather than smoothing it.
Especially with a small brush.
![ezgif com-crop](https://user-images.githubusercontent.com/41335511/42866766-6c366730-8a65-11e8-95b5-4ef5d316c09f.gif)

**Comment by Sinhouse85** [comment]
Ok, from further testing it appears that is an other issue caused by scale.
Its fine when scaled down.

**Comment by carter2422** [comment]
Will test this shortly @vxlcoder!

**Comment by Sinhouse85** [comment]
@vxlcoder 
Sure thing. I will report back.

**Comment by Sinhouse85** [comment]
@vxlcoder 
Ok. well its not destroying the mesh anymore. 
However, it's super slow to relax, ie really low relax strength.
Also it keeps throwing low fps warnings when relaxing.

**Comment by carter2422** [comment]
Unfortunately these changes have resulted in dramatically worse results. The strength issue that @Sinhouse85 mentioned appears to have been fixed but now it's causing haphazard flipping of faces for seemingly no rhyme or reason.

GIF: https://cl.ly/t6jF

**Comment by migero** [comment]
same here https://i.gyazo.com/8c075dfd4723192e509de792062e55ba.gif

**Comment by Hedrus** [comment]
The relax is not working in the latest version or previous versions I have been using.  I am on Blender 2.82 and am using RetoopoFlow_v3.0-beta_Blender-2.8x

When I use relax rather than the faces evening out the verts either scatter all over the place or they overlap badly and become less even.  Also it is quite slow.  
![image](https://user-images.githubusercontent.com/66180453/83336788-44b02a00-a284-11ea-91cd-311952196bad.png)

**Comment by MaximStale** [comment]
Yeah, it often happens, I try to use relax and tweak tool simultaneously

**Comment by jlampel** [comment]
I believe these issues have been addressed and the original problem of flipping faces has since been solved. There are still some known issues that we will be working on: [Relax Open Issues](https://github.com/CGCookie/retopoflow/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22tool%3A%20relax%22)

If you still have problems with Relax besides those, please feel free to open a new issue. This one is just too old and not very specific.
